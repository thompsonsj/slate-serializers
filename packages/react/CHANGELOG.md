# Changelog

## [2.3.0](https://github.com/thompsonsj/slate-serializers/compare/react-v2.2.3...react-v2.3.0) (2025-05-16)


### Features

* **react:** flatten config ([#191](https://github.com/thompsonsj/slate-serializers/issues/191)) ([761c651](https://github.com/thompsonsj/slate-serializers/commit/761c651ad49fda360b416ab561b9aac72ed7aa8d))
