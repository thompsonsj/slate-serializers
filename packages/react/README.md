# react

This library was generated with [Nx](https://nx.dev).

## Building

Run `nx build react` to build the library.

## Running unit tests

Run `nx test react` to execute the unit tests via [Jest](https://jestjs.io).
