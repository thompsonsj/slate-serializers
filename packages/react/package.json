{
  "name": "@slate-serializers/react",
  "version": "0.6.1",
  "main": "./index.js",
  "types": "./index.d.ts",
  "exports": {
    ".": {
      "import": "./index.mjs",
      "require": "./index.js"
    }
  }
}
