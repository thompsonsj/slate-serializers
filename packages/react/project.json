{
  "name": "react",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/react/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/react",
        "tsConfig": "packages/react/tsconfig.lib.json",
        "project": "packages/react/package.json",
        "entryFile": "packages/react/src/index.ts",
        "external": ["react", "react-dom", "react/jsx-runtime"],
        "rollupConfig": "@nx/react/plugins/bundle-rollup",
        "compiler": "babel",
        "assets": [
          {
            "glob": "packages/react/README.md",
            "input": ".",
            "output": "."
          }
        ],
        "updateBuildableProjectDepsInPackageJson": true,
        "buildableProjectDepsInPackageJsonType": "dependencies"
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/react/jest.config.ts"
      }
    }
  }
}
