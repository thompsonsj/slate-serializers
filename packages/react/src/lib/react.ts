export function react(): string {
  return 'react';
}
