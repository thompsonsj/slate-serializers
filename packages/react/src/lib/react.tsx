// Serializers
export { SlateToReact } from './serializers'

// Configuration objects
export type { Config as SlateToReactConfig } from './config/types'
export { config as slateToReactConfig } from './config/default'
export { config as payloadSlateToReactConfig } from './config/payload'
export { config as slateDemoSlateToReactConfig } from './config/slateDemo'
