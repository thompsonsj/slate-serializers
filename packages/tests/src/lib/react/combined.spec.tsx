import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { combinedFixtures, elementFixtures, textFixtures } from '../tests'
import { SlateToReact } from '@slate-serializers/react'

describe('Slate JSON to React transforms', () => {
  describe('Element tags', () => {
    const fixtures = elementFixtures
    for (const fixture of fixtures) {
      it(`${fixture.name}`, () => {
        const tree = render(<SlateToReact node={fixture.slate} />)
        expect(tree.container).toMatchSnapshot()
      })
    }
  })
  describe('Text tags', () => {
    const fixtures = textFixtures
    for (const fixture of fixtures) {
      it(`${fixture.name}`, () => {
        const tree = render(<SlateToReact node={fixture.slate} />)
        expect(tree.container).toMatchSnapshot()
      })
    }
  })
  describe('Combined', () => {
    const fixtures = combinedFixtures
    for (const fixture of fixtures) {
      it(`${fixture.name}`, () => {
        const tree = render(<SlateToReact node={fixture.slateOriginal} />)
        expect(tree.container).toMatchSnapshot()
      })
    }
  })
})
