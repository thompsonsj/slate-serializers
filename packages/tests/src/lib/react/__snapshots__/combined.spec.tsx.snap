// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Slate JSON to React transforms Combined larger document 1 1`] = `
<div>
  <h2>
    Heading 2
  </h2>
  A regular paragraph.
  A 
  <strong>
    paragraph
  </strong>
   with 
  <strong>
    various
  </strong>
   text 
  <s>
    marks
  </s>
  .
  <h3>
    Heading 3
  </h3>
  <ul>
    <li>
      <strong>
        Item 1
      </strong>
      : From a list
    </li>
    <li>
      <strong>
        Item 2
      </strong>
      : From a list
    </li>
  </ul>
  <a
    href="https://docs.slatejs.org/"
  >
    Find out more about Slate
  </a>
  .
  <h4>
    Heading 4
  </h4>
</div>
`;

exports[`Slate JSON to React transforms Combined links nested in an unordered list 1`] = `
<div>
  <ul>
    <li>
      <a
        href="https://github.com/thompsonsj/slate-serializers"
        target="_blank"
      >
        slate-serializers | GitHub
      </a>
    </li>
    <li>
      <a
        href="https://github.com/thompsonsj/slate-serializers"
        target="_blank"
      >
        slate-serializers | GitHub
      </a>
    </li>
  </ul>
</div>
`;

exports[`Slate JSON to React transforms Combined text with inline link 1`] = `
<div>
  Some text before an inline link 
  <a
    href="https://github.com/thompsonsj/slate-serializers"
    target="_blank"
  >
    slate-serializers | GitHub
  </a>
  .
</div>
`;

exports[`Slate JSON to React transforms Element tags blockquote 1`] = `
<div>
  <blockquote>
    Blockquote
  </blockquote>
</div>
`;

exports[`Slate JSON to React transforms Element tags headings 1`] = `
<div>
  <h1>
    Heading 1
  </h1>
  <h2>
    Heading 2
  </h2>
  <h3>
    Heading 3
  </h3>
  <h4>
    Heading 4
  </h4>
  <h5>
    Heading 5
  </h5>
  <h6>
    Heading 6
  </h6>
</div>
`;

exports[`Slate JSON to React transforms Element tags link 1`] = `
<div>
  <a
    href="https://github.com/thompsonsj/slate-serializers"
  >
    Slate Serializers | GitHub
  </a>
</div>
`;

exports[`Slate JSON to React transforms Element tags nested unordered list 1`] = `
<div>
  <ul>
    <li>
      Item 1
      <ul>
        <li>
          Nested item 1
        </li>
        <li>
          Nested item 2
        </li>
      </ul>
    </li>
    <li>
      Item 2
    </li>
  </ul>
</div>
`;

exports[`Slate JSON to React transforms Element tags ordered list 1`] = `
<div>
  <ol>
    <li>
      Item 1
    </li>
    <li>
      Item 2
    </li>
  </ol>
</div>
`;

exports[`Slate JSON to React transforms Element tags paragraph 1`] = `
<div>
  <p>
    Paragraph 1
  </p>
</div>
`;

exports[`Slate JSON to React transforms Element tags unordered list 1`] = `
<div>
  <ul>
    <li>
      Item 1
    </li>
    <li>
      Item 2
    </li>
  </ul>
</div>
`;

exports[`Slate JSON to React transforms Text tags idiomatic text 1`] = `
<div>
  <i>
    Idiomatic Text
  </i>
</div>
`;

exports[`Slate JSON to React transforms Text tags pre 1`] = `
<div>
  <pre>
    Pre
  </pre>
</div>
`;

exports[`Slate JSON to React transforms Text tags strikethrough 1`] = `
<div>
  <s>
    Strikethrough
  </s>
</div>
`;

exports[`Slate JSON to React transforms Text tags strong 1`] = `
<div>
  <strong>
    Bold
  </strong>
</div>
`;

exports[`Slate JSON to React transforms Text tags strong and idiomatic text 1`] = `
<div>
  <strong>
    Strong and Idiomatic Text
  </strong>
</div>
`;

exports[`Slate JSON to React transforms Text tags strong and unarticulated annotation 1`] = `
<div>
  <strong>
    Strong and Unarticulated Annotation
  </strong>
</div>
`;

exports[`Slate JSON to React transforms Text tags strong in paragraph 1`] = `
<div>
  <p>
    <strong>
      Bold
    </strong>
  </p>
</div>
`;

exports[`Slate JSON to React transforms Text tags strong mid-sentence 1`] = `
<div>
  Some 
  <strong>
    bold text
  </strong>
   in a sentence.
</div>
`;

exports[`Slate JSON to React transforms Text tags unarticulated annotation 1`] = `
<div>
  <u>
    Unarticulated Annotation
  </u>
</div>
`;
